app/merger.go

```diff
@@ -48,14 +48,15 @@ func (smartMerger) Merge(reports []report.Report) report.Report {
 	case 1:
 		return reports[0]
 	}
-	c := make(chan report.Report, l)
-	for _, r := range reports {
-		c <- r
+	c := make(chan *report.Report, l)
+	for i := range reports {
+		c <- &reports[i]
 	}
 	for ; l > 1; l-- {
-		go func(left, right report.Report) {
-			c <- left.Merge(right)
+		go func(left, right *report.Report) {
+			r := left.Merge(*right)
+			c <- &r
 		}(<-c, <-c)
 	}
-	return <-c
+	return *<-c
```

Comment:
In slack I had proposed the following as a simpler change, which is blissfully pointer-free:
```diff
diff --git a/app/merger.go b/app/merger.go
index be4d149..94231ba 100644
--- a/app/merger.go
+++ b/app/merger.go
@@ -53,9 +53,10 @@ func (smartMerger) Merge(reports []report.Report) report.Report {
               c <- r
       }
       for ; l > 1; l-- {
-               go func(left, right report.Report) {
+               left, right := <-c, <-c
+               go func() {
                       c <- left.Merge(right)
-               }(<-c, <-c)
+               }()
       }
       return <-c
}
```